### DISCUSSION TITLE
How do I make my plugin automatically call the correct setup function?

### DISCUSSION THREAD

**Discussion by codymikol** [LazyVim/LazyVim#2358] [Q&A] [answered]
The docs say that LazyVim will automatically require "MAIN", how does lazyvim determine what MAIN is?

**Comment by dpetka2001** [comment, marked as answer]
It doesn't. `lazy.nvim` the package manager determines what MAIN is. You should read its code for how it determines that, as I couldn't find anything more than a reference in its docs and I haven't looked into that specific thing myself.

**Comment by codymikol** [comment]
Thank you, I didn't realize there were two different repos, appreciate you!
